agimus_controller/agimus_controller/robot_model_factory/robot_model_factory.py

```diff
@@ -0,0 +1,144 @@
+from typing import Union
+
+try:
+    from typing import Self
+except ImportError:
+    from typing_extensions import Self
+from dataclasses import dataclass
+from copy import deepcopy
+import pinocchio as pin
+import numpy as np
+from pathlib import Path
+from .obstacle_params_parser import ObstacleParamsParser
+
+
+@dataclass
+class RobotModelParameters:
+    q0_name = str()
+    free_flyer = False
+    locked_joint_names = []
+    urdf = Path()
+    srdf = Path()
+    collision_as_capsule = False
+    self_collision = False
+
+
+class RobotModelFactory:
+    """Complete model of the robot."""
+
+    _model = pin.Model()
+    """ Complete model of the robot with visualization meshes. """
+    _cmodel = pin.GeometryModel()
+    """ Complete model of the robot with collision meshes. """
```

Comment:
From my understanding, there is no difference between reduced and complete collision model but it needs to be verified; because reduced model is simply without all the joints actuated right? 

We can use the convention: 

cmodel stands for: collision model convexified
vmodel stands for: visual model (ie collision model but with meshes instead) 

What do y'all think?

Reply:
hum good question, I do not know.
I guess we could remove the notion of "reduced model" in the code as we only use the reduced model anyway.

Reply:
I'm not sure, keeping the ability to close the gripper in some cases will be helpful. 
I believe that have both reduced and full actuation model is important.
I was mentionning the collisions models